node/lib/util/read_repo_ast_util.js

```diff
@@ -324,6 +336,35 @@ exports.readRAST = co.wrap(function *(repo) {
                 }
             }
         }
+
+        // Check for changes to submodules not reflected in the index (other
```

Comment:
ReadRAST is already hundreds of lines long. Can we break this out into a separate function?